metakb/database.py

```diff
@@ -0,0 +1,371 @@
+
+        descr_keys = _create_keys_string(descriptor, ('id', 'label',
+                                                      'description', 'xrefs',
+                                                      'alternate_labels'))
+
+        query = f'''
+        MERGE (descr:{descr_type} {{ {descr_keys} }})
+        MERGE (value:{value_type} {{ id:$value_id }})
+        MERGE (descr) -[:DESCRIBES]-> (value)
+        '''
+        try:
+            tx.run(query, **descriptor)
+        except ServiceUnavailable as exception:
+            logging.error(f"Failed to add Descriptor object\nQuery: {query}\n"
+                          f"Descriptor: {descriptor}")
+            raise exception
+
+    @staticmethod
+    def _add_variation_descriptor(tx, descriptor_in: Dict):
+        """Add variant descriptor object to DB.
+        :param Dict descriptor_in: must include a `value_id` field and a
+            `value` object containing `type`, `state`, and `location` objects.
+        """
+        descriptor = descriptor_in.copy()
+
+        # prepare value properties
+        value_type = descriptor['value']['type']
+        descriptor['value_state'] = json.dumps(descriptor['value']['state'])
+        location = descriptor['value']['location']
+        descriptor['value_location_type'] = location['type']
+        descriptor['value_location_sequence_id'] = location['sequence_id']
+        descriptor['value_location_interval_start'] = \
+            location['interval']['start']
+        descriptor['value_location_interval_end'] = location['interval']['end']
+        descriptor['value_location_interval_type'] = \
+            location['interval']['type']
+
+        # prepare descriptor properties
+        expressions = descriptor.get('expressions')
+        if expressions:
+            for expression in expressions:
+                syntax = expression['syntax'].split(':')[1]
+                key = f"expressions_{syntax}"
+                if key in descriptor:
+                    descriptor[key].append(expression['value'])
+                else:
+                    descriptor[key] = [expression['value']]
+
+        nonnull_keys = [_create_keys_string(descriptor,
+                                            ('id', 'label', 'description',
+                                             'xrefs', 'alternate_labels',
+                                             'structural_type',
+                                             'molecule_context',
+                                             'expressions_transcript',
+                                             'expressions_genomic',
+                                             'expressions_protein',
+                                             'ref_allele_seq'))]
+
+        # handle extensions
+        variant_groups = None
+        extensions = descriptor.get('extensions')
+        if extensions:
+            for ext in extensions:
+                name = ext['name']
+                if name == 'variant_groups':
```

Comment:
This is now `variant_group`: https://github.com/cancervariants/metakb/blob/issue-56/metakb/transform/civic.py#L502